overlord/snapstate/backend/snapdata.go

```diff
@@ -108,8 +109,14 @@ func copySnapData(oldSnap, newSnap *snap.Info) (err error) {
 	return nil
 }
 
-// Lowlevel copy the snap data (but never override existing data)
+// Lowlevel copy the snap data
 func copySnapDataDirectory(oldPath, newPath string) (err error) {
+	if osutil.FileExists(newPath) {
```

Comment:
This should probably be osutil.IsDirectory instead. Feels slightly awkward that FileExists returns true here. Either a misnomer or a bug.